lib/build/geo_feature_builder.dart

```diff
@@ -0,0 +1,201 @@
+import 'dart:core';
+import 'dart:io';
+import 'dart:convert';
+import 'package:build/build.dart';
+import 'package:code_builder/code_builder.dart';
+import 'package:dart_style/dart_style.dart';
+import 'package:recase/recase.dart';
+
+class GeoFeatureBuilder implements Builder{
+
+  final BuilderOptions options;
+
+  GeoFeatureBuilder(this.options);
+  
+  @override
+  Map<String, List<String>> get buildExtensions => {
+    options.config['source_file'] : [ options.config['output_file'] ],
+  };
+
+  @override
+  Future<void> build(BuildStep buildStep) async {
+    final AssetId inputId = buildStep.inputId;
+    try {
+        final geoJSONData = await _readFile(File(inputId.path));
+        final propertiesToExtract = options.config['extract_properties'].map<String,String>(
+            (key, value) => MapEntry<String, String>(key.toString(), value.toString())
+        );
+        final geoFeatureList = buildGeoFeatureListInstance(geoJSONData, propertiesToExtract);
+        final code = generateFileCode(geoFeatureList,propertiesToExtract);
+        final outputId = AssetId(inputId.package, inputId.path
+          .replaceFirst(options.config['source_file'],options.config['output_file'])
+        );
+        await buildStep.writeAsString(outputId, code);
+    } catch (e) {
+        print('Failed to generate class: $e');
+    }
+  }
+
+  String generateFileCode(LiteralListExpression geoFeatureList, Map<String, String> propertiesToExtract){
+    final geoFeatureClass = buildGeoFeatureClass(propertiesToExtract);
+    final geoCoderClass = buildGeoCoderClass(geoFeatureList);
+    final library = Library((b) => b
+      ..directives.add(Directive.import('package:offline_geocoder/offline_geocoder.dart'))
+      ..directives.add(Directive.import('package:latlong2/latlong.dart'))
+      ..body.addAll([geoFeatureClass, geoCoderClass]
+    ));
+    final dartfmt = DartFormatter();
+    return dartfmt.format('${library.accept(DartEmitter.scoped())}').toString();
+  }
+
+  Future<Map<String, dynamic>> _readFile(File inputFile) async {
+    if (!inputFile.existsSync()) {
+      throw FileSystemException('The file ${inputFile.path} does not exist');
+    }
+    String data;
+    try {
+      data = await inputFile.readAsString();
+      return json.decode(data) as Map<String, dynamic>;
+    } catch (e) {
+      throw FileSystemException('The file $e can not be read');
+    }
+  }
+
+  LiteralListExpression buildGeoFeatureListInstance(Map<String, dynamic> geoJSONData, Map<String, String> propertiesToExtract) {
+    final features = geoJSONData['features'].cast<Map<String, dynamic>>(); 
+    final geoFeatureRefer = refer('GeoFeature');
+    final List<Expression> geoFeatureList = [];
+    for (final featureData in features) {
+      final List<Expression> geoFeatureParameters = [];
+      if (featureData.containsKey('properties')) {
+        final featureProperties = featureData['properties'];
```

Comment:
Change to:
```dart
    final featureProperties = featureData['properties'];
    if (featureProperties != null) {
```
Edit:
At the end I think we should even remove this check since we anyway rely on the fact that the properties defined in "extract" are present in the geojson source.

Perhaps you can turn it into an assert.